dev/Interop/StoragePickers/PickerCommon.cpp

```diff
@@ -142,116 +107,27 @@ namespace PickerCommon {
-            buffer.push_back(L"");
-            buffer.push_back(ext);
-
-            allFilesExtensionList.reserve(allFilesExtensionList.length() + ext.size() + 1);
-            allFilesExtensionList += ext;
-            allFilesExtensionList += L";";
-        }
-
-        if (!allFilesExtensionList.empty())
-        {
-            allFilesExtensionList.pop_back(); // Remove the last semicolon
-        }
-
-        if (filters.Size() == 0)
-        {
-            // when filters not defined, set filter to All Files *.*
-            buffer.push_back(L"");
-            buffer.push_back(L"*");
-        }
-        else if (filters.Size() == 1 && allFilesExtensionList == L"*")
-        {
-            // when there're only one filter "*", set filter to All Files *.* (override the values pushed above)
-            buffer[0] = L"";
-            buffer[1] = L"*";
-            resultSize = 1;
-        }
-        else
-        {
-            buffer.push_back(L"");
-            buffer.push_back(allFilesExtensionList.c_str());
-        }
-
-        std::vector<COMDLG_FILTERSPEC> result{ resultSize };
-        for (size_t i = 0; i < resultSize; i++)
-        {
-            result.at(i) = { buffer.at(i * 2).c_str(), buffer.at(i * 2 + 1).c_str() };
-        }
-
-        return result;
-    }
-
-    /// <summary>
-    /// Capture and processing pickers filter inputs and convert them into Common Item Dialog's accepting type, for FileSavePicker
-    /// </summary>
-    /// <param name="buffer">temp buffer to hold dynamically transformed strings</param>
-    /// <param name="filters">winrt style filters</param>
-    /// <returns>result Coomon Item Dialog style filters, note only raw pointers here,
-    /// they are valid up to lifetime of buffer
-    /// </returns>
-    std::vector<COMDLG_FILTERSPEC> CaptureFilterSpec(std::vector<winrt::hstring>& buffer, winrt::Windows::Foundation::Collections::IMapView<winrt::hstring, winrt::Windows::Foundation::Collections::IVector<hstring>> filters)
-    {
-        std::vector<COMDLG_FILTERSPEC> result(filters.Size());
-        buffer.clear();
-        buffer.reserve(filters.Size() * static_cast<size_t>(2));
-
-        for (const auto& filter : filters)
-        {
-            buffer.push_back(filter.Key());
-            auto extensionList = JoinExtensions(filter.Value().GetView());
-            buffer.push_back(extensionList);
-        }
-        for (size_t i = 0; i < filters.Size(); i++)
-        {
-            result.at(i) = { buffer.at(i * 2).c_str(), buffer.at(i * 2 + 1).c_str() };
-        }
-
-        if (result.empty())
-        {
-            result.push_back({ L"", L"*.*" });
-        }
-        return result;
-    }
-
-    void PickerParameters::ConfigureDialog(winrt::com_ptr<IFileDialog> dialog)
+    void ConfigureDialog(winrt::com_ptr<IFileDialog> dialog, const StoragePickersImpl::PickerParameters & parameters, winrt::Microsoft::Windows::Storage::Pickers::PickerLocationId pickerLocationId)
```

Comment:
I think this would be better still as a member of PickerParameters. I'm guessing it is here because separation of PickerParameters into a lib made it more complicated to be aware of IFileDialog and the other helpers here in PickerCommon.